internal/metrics/types.go

```diff
@@ -190,3 +194,42 @@ type ProviderFactory interface {
 	Name() string
 	Build(uri *url.URL) (MetricsSourceProvider, error)
 }
+
+type ConfigurabeMetricsSourceProvider interface {
+	Configure(uri *url.URL)
+}
+
+//DefaultMetricsSourceProvider handle the common providers configuration
+type DefaultMetricsSourceProvider struct {
+	collectionInterval time.Duration
+	timeOut            time.Duration
+}
+
+// CollectionInterval return the provider collection interval configuration
+func (dp *DefaultMetricsSourceProvider) CollectionInterval() time.Duration {
+	return dp.collectionInterval
+}
+
+// TimeOut return the provider timeout configuration
+func (dp *DefaultMetricsSourceProvider) TimeOut() time.Duration {
+	return dp.timeOut
+}
+
+// Configure the 'collectionInterval' and 'timeOut' values
+func (dp *DefaultMetricsSourceProvider) Configure(uri *url.URL) {
+	vals := uri.Query()
+	dp.collectionInterval = parseDuration(vals, "collectionInterval", time.Duration(10*time.Second))
+	dp.timeOut = parseDuration(vals, "timeOut", time.Duration(10*time.Second))
+}
+
+func parseDuration(vals url.Values, prop string, def time.Duration) time.Duration {
```

Comment:
Can move this here: https://github.com/wavefrontHQ/wavefront-kubernetes-collector/blob/master/internal/flags/utils.go